src/features/courses/course-route.ts

```diff
@@ -1337,11 +1337,16 @@ router.post('/question/editor/read',
     // This is due to a typescript issue where the type mismatches extractMap
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     asyncHandler(async (req: RederlyExpressRequest<ReadQuestionRequest.params, unknown, ReadQuestionRequest.body, ReadQuestionRequest.query>, _res: Response, next: NextFunction) => {
+        const sourceFilePath = req.body.filePath;
+        // TODO if we use this regex elsewhere we should centralize
+        // This should also be in the joi validation but due to time putting it here to handle the frontend
+        if (!/^(:?private\/|Contrib\/|webwork-open-problem-library\/Contrib\/|Library\/|webwork-open-problem-library\/OpenProblemLibrary\/)\S+\.pg$/.test(sourceFilePath)) {
```

Comment:
I put `VALID_PG_PATH_REGEX` in renderer-helper

Reply:
Yeah I saw that but I think it was a little different, I should switch this and will do so in the future if I can